scripts/generate-custom-properties.js

```diff
@@ -109,12 +89,63 @@ Object.entries(definitions.plainTokens).forEach(([key, value]) => {
   tokens[key] = value;
 });
 
+// Write files
+const printJson = data =>
+  JSON.stringify(
+    Object.entries(data).reduce(
+      (acc, [key, value]) => ({
+        ...acc,
+        [`--${key}`]: value,
+      }),
+      {}
+    )
+  );
+
+const printJavaScript = data => `
+/*
+  THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+
+  This file is created by the 'scripts/generate-custom-properties.js' script.
+  The variables should be updated in 'materials/internals/definition.yaml'.
+*/
+
+${Object.entries(data)
+  .map(([key, value]) => `export const ${camelCase(key)} = "${value}";`)
```

Comment:
This is for the new file yes

Reply:
Why do we generate loads of exports instead of building one object and default exporting that. This I would assume loads to loads of import * as vars without ever really intendig to import one var at a time? feels like we would just read the json and transform keys?

Reply:
Sorry I don't get your point. Having named exports helps with things like typeahead

<img width="829" alt="screenshot 2019-01-22 at 21 51 54" src="https://user-images.githubusercontent.com/1110551/51564592-fe78d880-1e8f-11e9-84f3-fc81db35d1ef.png">

and treeshaking, as opposed to exporting one single object.

Reply:
I understand the tree-shaking argument but find it out of proportion for a JSON or object currently not too big but likely to grow. Furthermore, all tokens are used within UIKit anyway so to my understanding there is little to hale. Anyway you‘re saying that tree-shaking now works when importing with * as? Autocompletion would also be totally borked when exporting an object (which in a possible future will be typed)?

Anyway I don‘t care too much. Just thought an object would be simpler if we always alias the * import to vars anyway.

Reply:
We can export it as an object, I don’t mind as long as we get the benefits such as typeahead 😇

As for the `import * as`, I’m not so sure anymore if that tree shakes, as opposed of importing like `import { colorBlack }`...good point!

I’ll refactor the script then. Thanks for the feedback 👌

Reply:
> all tokens should be used anyway

Depends which components you actually use at the end, right? If I only use the Text input, I won’t be needing the majority of the tokens. But anyway 😄

Reply:
I'll end the argument here. Thanks for the insights.

Reply:
🤔 true

Reply:
I agree but it's all a bit of a theoretical case (for sure only for now). We're not sure how well tree-shaking works. We don't have many consumers for the library yet and no numbers on it.